frame/democracy/src/tests/fast_tracking.rs

```diff
@@ -81,6 +85,60 @@ fn instant_referendum_works() {
 	});
 }
 
+#[test]
+fn instant_next_block_referendum_backed() {
+	new_test_ext().execute_with(|| {
+		// arrange
+		let start_block_number = 10;
+		let majority_origin_id = 3;
+		let instant_origin_id = 6;
+		let voting_period = 1;
+		let proposal_hash = set_balance_proposal_hash_and_note(2);
+		let delay = 2; // has no affect on test
+
+		// init
+		System::set_block_number(start_block_number);
+		InstantAllowed::set(true);
+
+		// propose with majority origin
+		assert_ok!(Democracy::external_propose_majority(
+			Origin::signed(majority_origin_id),
+			proposal_hash
+		));
+
+		// fast track with instant origin and voting period pointing to the next block
+		assert_ok!(Democracy::fast_track(
+			Origin::signed(instant_origin_id),
+			proposal_hash,
+			voting_period,
+			delay
+		));
+
+		// fetch the status of the only referendum at index 0
+		assert_eq!(
+			Democracy::referendum_status(0),
+			Ok(ReferendumStatus {
+				end: start_block_number + voting_period,
+				proposal_hash,
+				threshold: VoteThreshold::SimpleMajority,
+				delay,
+				tally: Tally { ayes: 0, nays: 0, turnout: 0 },
+			})
+		);
+
+		// referendum expected to be backed with the start of the next block
```

Comment:
```suggestion
		// referendum expected to be baked with the start of the next block
```

Reply:
This should actually be `backed at`, not `baked with`.

Reply:
It's referring to baking (checking if approved and scheduling dispatch) at the beginning of the expiring block. The bug this is fixing was not that a 0-length fast track had no _backing_, it's that _baking_ takes place at the start of a block, so zero-length referenda never get baked.

https://github.com/paritytech/substrate/blob/60b945caca440fdf6a7be96b7dc0db54923f039d/frame/democracy/src/lib.rs#L1783

Reply:
Got it, thank you for the context :+1: